src/tests/ftest/nvme/nvme_pool_extend.py

```diff
@@ -0,0 +1,242 @@
+        key = "".join([oclass, api, str(test[0])])
+        manager.job.dfs_cont.update(self.container_info[key])
+        env = ior_cmd.get_default_env(str(manager))
+        manager.assign_hosts(self.hostlist_clients, self.workdir, None)
+        manager.assign_processes(processes)
+        manager.assign_environment(env, True)
+
+        # run IOR Command
+        try:
+            manager.run()
+        except CommandFailure as _error:
+            results.put("FAIL")
+
+    def run_ior_thread(self, action, oclass, api, test):
+        """[summary]
+        Args:
+            action (str): Start the IOR thread with Read or
+                          Write
+            oclass (str): IOR object class
+            API (str): IOR API
+            test (list): IOR test sequence
+            flags (str): IOR flags
+        """
+        if action == "Write":
+            flags = self.ior_w_flags
+        else:
+            flags = self.ior_r_flags
+
+        # Add a thread for these IOR arguments
+        process = threading.Thread(target=self.ior_thread,
+                                   kwargs={"pool": self.pool,
+                                           "oclass": oclass,
+                                           "api": api,
+                                           "test": test,
+                                           "flags": flags,
+                                           "results":
+                                           self.out_queue})
+        # Launch the IOR thread
+        process.start()
+        # Wait for the thread to finish
+        process.join()
+
+    def run_nvme_pool_extend(self, num_pool):
+        """Run Pool Extend
+            Args:
+             int : total pools to create for testing purposes.
+        """
+        # Create a pool
+        pool = {}
+        total_servers = len(self.hostlist_servers) * 2
+
+        # Extend one of the ranks (or server)
+        # rank index starts from zero
+        rank = total_servers
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            for oclass, api, test in product(self.ior_dfs_oclass,
+                                             self.ior_apis,
+                                             self.ior_test_sequence):
+                self.run_ior_thread("Write", oclass, api, test)
+
+                scm_size = self.pool.scm_size
+                nvme_size = self.pool.nvme_size
```

Comment:
The use of `scm_size` and `nvme_size` aren't clear.

Reply:
Changed it to nvme_pool_size, scm_pool_size